src/utils/uploadImage.ts

```diff
@@ -0,0 +1,103 @@
+import { setScreenshot } from "@/store/reducer/canvasControlSlice";
+import { switchRuler } from "@/store/reducer/buttonToggleSlice";
+import { resetAll } from "@/store/reducer/canvasControlSlice";
+import { Dispatch } from "@reduxjs/toolkit";
+import { RefObject } from "react";
+import Konva from "konva";
+
+export const upLoadScreenshot = async (screenshot: string, toast: any) => {
+  const { default: AWS } = await import(/* webpackChunkName: "aws-sdk" */ "aws-sdk");
+  const { nanoid } = await import("nanoid");
+
+  const albumBucketName = process.env.NEXT_PUBLIC_ALBUM_BUCKET_NAME as string;
+  const bucketRegion = process.env.NEXT_PUBLIC_BUCKET_REGION as string;
+  const IdentityPoolId = process.env.NEXT_PUBLIC_IDENTITY_POOL_ID as string;
+  const albumName = process.env.NEXT_PUBLIC_ALBUM_NAME as string;
+  const albumPhotosKey = encodeURIComponent(albumName) + "/";
+
+  const base64 = await fetch(screenshot);
+  const blob = await base64.blob();
+  const imgFile = new File([blob], "default", { type: "image/png" });
+  let imageUrl = "" as string;
+
+  AWS.config.update({
+    region: bucketRegion,
+    credentials: new AWS.CognitoIdentityCredentials({
+      IdentityPoolId: IdentityPoolId,
+    }),
+  });
+  const photoKey = albumPhotosKey + nanoid() + ".png";
+  const upload = new AWS.S3.ManagedUpload({
+    params: {
+      Bucket: albumBucketName,
+      Key: photoKey,
+      Body: imgFile,
+    },
+  });
+
+  const promise = upload.promise();
+  await promise.then(
+    function (data) {
+      imageUrl = data.Location;
+    },
+    function (err) {
+      return toast({
+        title: `Image update failure: ${err.message}.`,
+        description: "Try again or contact IT support",
+        status: "error",
+        duration: 9000,
+        isClosable: true,
+        position: "top",
+      });
+    }
```

Comment:
It is working but all these can be simplified a little if you want.

Reply:
solved!